src/Microsoft.Health.Fhir.SqlServer/Features/Storage/Registry/SqlServerSearchParameterStatusDataStore.cs

```diff
@@ -0,0 +1,120 @@
+            _updateSearchParamsTvpGenerator = updateSearchParamsTvpGenerator;
+            _filebasedSearchParameterStatusDataStore = filebasedRegistry.Invoke();
+            _schemaInformation = schemaInformation;
+        }
+
+        // TODO: Make cancellation token an input.
+        public async Task<IReadOnlyCollection<ResourceSearchParameterStatus>> GetSearchParameterStatuses()
+        {
+            // If the search parameter table in SQL does not yet contain status columns
+            if (_schemaInformation.Current < SchemaVersionConstants.SearchParameterStatusSchemaVersion)
+            {
+                // Get status information from file.
+                return await _filebasedSearchParameterStatusDataStore.GetSearchParameterStatuses();
+            }
+
+            using (IScoped<SqlConnectionWrapperFactory> scopedSqlConnectionWrapperFactory = _scopedSqlConnectionWrapperFactory())
+            using (SqlConnectionWrapper sqlConnectionWrapper = await scopedSqlConnectionWrapperFactory.Value.ObtainSqlConnectionWrapperAsync(CancellationToken.None, true))
+            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                VLatest.GetSearchParamStatuses.PopulateCommand(sqlCommandWrapper);
+
+                var parameterStatuses = new List<ResourceSearchParameterStatus>();
+
+                using (SqlDataReader sqlDataReader = await sqlCommandWrapper.ExecuteReaderAsync(CommandBehavior.SequentialAccess, CancellationToken.None))
+                {
+                    while (await sqlDataReader.ReadAsync())
+                    {
+                        (string uri, string stringStatus, DateTimeOffset? lastUpdated, bool? isPartiallySupported) = sqlDataReader.ReadRow(
+                            VLatest.SearchParam.Uri,
+                            VLatest.SearchParam.Status,
+                            VLatest.SearchParam.LastUpdated,
+                            VLatest.SearchParam.IsPartiallySupported);
+
+                        var status = Enum.Parse<SearchParameterStatus>(stringStatus, true);
+
+                        var resourceSearchParameterStatus = new ResourceSearchParameterStatus()
+                        {
+                            Uri = new Uri(uri),
+                            Status = status,
+                            IsPartiallySupported = (bool)isPartiallySupported,
+                            LastUpdated = (DateTimeOffset)lastUpdated,
+                        };
+
+                        parameterStatuses.Add(resourceSearchParameterStatus);
+                    }
+                }
+
+                return parameterStatuses;
+            }
+        }
+
+        // TODO: Make cancellation token an input.
+        public async Task UpsertStatuses(IEnumerable<ResourceSearchParameterStatus> statuses)
+        {
+            EnsureArg.IsNotNull(statuses, nameof(statuses));
+
+            if (!statuses.Any())
```

Comment:
>statuses.Any() [](start = 17, length = 14)

`IEnumerable` is a promise to provide something on request.
Every time it's requested it's has to evaluate itself. 
If `List<>` passed it wouldn't calculate anything, and just enumerate list, but if instead of `List<>` some real `IEnumerable` passed which actually does calculation or issue network requests, it has to perform it for `Any` and  below for `statuses.ToList()`

So I would either cache `IEnumerable` into `List` collection immediately and then check Count here , and pass that list to `Generate`, or changed signature of function to actually accept `List<>`.

Reply:
Ah, thank you for the explanation. I modified the method to accept a `List` instead of `IEnumberable` in 307a244.